python/ray/serve/scripts.py

```diff
@@ -371,6 +372,17 @@ def deploy(config_file_name: str, address: str):
         "will be ignored if running a config file."
     ),
 )
+@click.option(
+    "--name",
+    required=False,
+    default=SERVE_DEFAULT_APP_NAME,
+    type=str,
+    help=(
+        "Name of the application. This should only be used "
+        "when running an application specified by import path and "
+        "will be ignored if running a config file."
+    ),
```

Comment:
Not really caused by this PR, but a general note on API design:

If you find yourself writing these types of "conditional" semantics (e.g., "argument ABC is only used if DEF is also provided"), it's usually a red flag that the interface could be better designed. For example, in this case it seems that we most likely should not be accepting both a config file and import path in the same command.

(cc @zcin for visibility as well)